src/protocols/bdx/BdxTransferSession.cpp

```diff
@@ -8,11 +8,16 @@
 
 #include <protocols/Protocols.h>
 #include <protocols/bdx/BdxMessages.h>
+#include <protocols/common/Constants.h>
+#include <support/BufferReader.h>
 #include <support/CodeUtils.h>
+#include <support/ReturnMacros.h>
 #include <system/SystemPacketBuffer.h>
 
 namespace {
-const uint8_t kBdxVersion = 0; ///< The version of this implementation of the BDX spec
+constexpr uint8_t kBdxVersion            = 0;         ///< The version of this implementation of the BDX spec
+constexpr size_t kStatusReportMinSize    = 2 + 4 + 2; ///< 16 bits for GeneralCode, 32 bits for ProtocolId, 16 bits for ProtocolCode
+constexpr uint16_t kStatusReportVendorId = 0;
```

Comment:
@andreilitvin - When BDX is putting together a StatusReport message, how should it set the vendorId part of the StatusReport?

Reply:
@andy31415

Reply:
StatusCodes in CHIP only use a protocolid since the protocolid is supposed to be a combination of vendor + vendor-specific-protocol.

I believe our define would be:

```
kProtocol_BDX                 = (kChipVendor_Common << 16) | 0x0003, // Bulk Data Exchange Protocol
```

so if you really need the vendorid, it is the common one, however for status codes you likely need only kProtocol_BDX without splitting into vendorid and others.

Reply:
Define taken from Protocols.h

Reply:
Oh, I understand now. I wasn't seeing the `(kChipVendor_Common << 16)` part of Protocol.h. I've already defined the BDX protocol ID that way in the PR. I just copied and pasted, I didn't realize the vendorID was already baked in :)